packages/compass/src/app/auto-connect.spec.ts

```diff
@@ -1,9 +1,52 @@
 import { expect } from 'chai';
 import sinon from 'sinon';
 import { loadAutoConnectInfo } from './auto-connect';
-import { exportConnections } from '@mongodb-js/connection-storage';
-import type { ConnectionInfo } from '@mongodb-js/connection-storage';
+import {
+  ConnectionStorage,
+  type ExportConnectionOptions,
+} from '@mongodb-js/connection-storage/main';
+import type { ConnectionInfo } from '@mongodb-js/connection-storage/renderer';
 import { ipcRenderer } from 'hadron-ipc';
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+
+const loadAutoConnectWithConnection = async (
+  connectPreferences: Record<string, unknown> = {},
+  connections: ConnectionInfo[] = [],
+  exportOptions: ExportConnectionOptions = {}
+) => {
+  const tmpDir = fs.mkdtempSync(
+    path.join(os.tmpdir(), 'connection-storage-tests')
+  );
+  fs.mkdirSync(path.join(tmpDir, 'Connections'));
+
+  ConnectionStorage['path'] = tmpDir;
+  await Promise.all(
+    connections.map((connectionInfo) =>
+      ConnectionStorage.save({ connectionInfo })
+    )
+  );
+
+  const fileContents = await ConnectionStorage.exportConnections({
+    options: exportOptions,
+  });
+
+  const fakeFs = { readFile: sinon.stub().resolves(fileContents) };
+  const deserializeConnections =
+    ConnectionStorage.deserializeConnections.bind(ConnectionStorage);
+
+  const fn = await loadAutoConnectInfo(
+    sinon.stub().resolves(connectPreferences),
+    fakeFs,
+    deserializeConnections
+  );
+
+  // clean up
+  fs.rmdirSync(tmpDir, { recursive: true });
```

Comment:
Do you want to wrap the part above in a `try`/`finally` to ensure that this always cleans up? (And if you like, feel free to use the promisified fs variants instead, afaict filesystem operations are used in async functions only anyway, and yes it doesn’t make much of a difference in tests, but it’s usually a good idea to follow the same rules in tests as regular source code if there’s no reason not to)